From 493704f6dbd7a979798eebd49aa54fe1b9583f64 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Mon, 26 Oct 2015 18:39:05 -0700
Subject: [PATCH 068/206] SnapdragonCamera: Add option to control antibanding
 in camcorder

The antibanding setting in camcorder is linked to the setting in camera
mode for better UX (if you change one, the other will change accordingly).

Change-Id: I98c1bc65472cf8eec530f5a7572f46458b0ab3fa
---
 res/values/qcomstrings.xml              |  4 ++--
 res/xml/video_preferences.xml           |  6 ++++++
 src/com/android/camera/VideoMenu.java   |  1 +
 src/com/android/camera/VideoModule.java | 23 +++++++----------------
 4 files changed, 16 insertions(+), 18 deletions(-)

diff --git a/res/values/qcomstrings.xml b/res/values/qcomstrings.xml
index ecdf116f2..93d574a4e 100755
--- a/res/values/qcomstrings.xml
+++ b/res/values/qcomstrings.xml
@@ -523,8 +523,8 @@
 
      <!-- Settings screen, Anti Banding entries -->
      <string name="pref_camera_antibanding_entry_0">Off</string>
-     <string name="pref_camera_antibanding_entry_1">50 Hz</string>
-     <string name="pref_camera_antibanding_entry_2">60 Hz</string>
+     <string name="pref_camera_antibanding_entry_1">50 Hz (Europe)</string>
+     <string name="pref_camera_antibanding_entry_2">60 Hz (USA)</string>
      <string name="pref_camera_antibanding_entry_3">Auto</string>
 
         <!-- Default color effect setting. -->
diff --git a/res/xml/video_preferences.xml b/res/xml/video_preferences.xml
index bcf52e2fb..3bc72acf9 100644
--- a/res/xml/video_preferences.xml
+++ b/res/xml/video_preferences.xml
@@ -184,6 +184,12 @@
             camera:title="@string/pref_camera_video_rotation_title"
             camera:entries="@array/pref_camera_video_rotation_entries"
             camera:entryValues="@array/pref_camera_video_rotation_entryvalues" />
+    <ListPreference
+            camera:key="pref_camera_antibanding_key"
+            camera:defaultValue="@string/pref_camera_antibanding_default"
+            camera:title="@string/pref_camera_antibanding_title"
+            camera:entries="@array/pref_camera_antibanding_entries"
+            camera:entryValues="@array/pref_camera_antibanding_entryvalues" />
     <ListPreference
             camera:key="pref_camera_zoom_key"
             camera:defaultValue="@string/pref_camera_zoom_default"
diff --git a/src/com/android/camera/VideoMenu.java b/src/com/android/camera/VideoMenu.java
index adf8040d7..b6a03b6d7 100755
--- a/src/com/android/camera/VideoMenu.java
+++ b/src/com/android/camera/VideoMenu.java
@@ -141,6 +141,7 @@ public class VideoMenu extends MenuController
                 CameraSettings.KEY_VIDEO_ENCODER,
                 CameraSettings.KEY_AUDIO_ENCODER,
                 CameraSettings.KEY_VIDEO_HDR,
+                CameraSettings.KEY_ANTIBANDING,
                 CameraSettings.KEY_POWER_MODE,
                 CameraSettings.KEY_VIDEO_ROTATION,
                 CameraSettings.KEY_VIDEO_CDS_MODE,
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 849263728..07471c436 100755
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -363,7 +363,6 @@ public class VideoModule implements CameraModule,
     private boolean mUnsupportedHFRVideoSize = false;
     private boolean mUnsupportedHSRVideoSize = false;
     private boolean mUnsupportedHFRVideoCodec = false;
-    private String mDefaultAntibanding = null;
     boolean mUnsupportedProfile = false;
 
     // This Handler is used to post message back onto the main thread of the
@@ -2423,23 +2422,15 @@ public class VideoModule implements CameraModule,
             }
         }
 
-        if (mDefaultAntibanding == null) {
-            mDefaultAntibanding = mParameters.getAntibanding();
-            Log.d(TAG, "default antibanding value = " + mDefaultAntibanding);
+        // Set anti banding parameter.
+        String antiBanding = mPreferences.getString(
+                 CameraSettings.KEY_ANTIBANDING,
+                 mActivity.getString(R.string.pref_camera_antibanding_default));
+        Log.v(TAG, "antiBanding value =" + antiBanding);
+        if (CameraUtil.isSupported(antiBanding, mParameters.getSupportedAntibanding())) {
+            mParameters.setAntibanding(antiBanding);
         }
 
-        if (disMode.equals("enable")) {
-            Log.d(TAG, "dis is enabled, set antibanding to auto.");
-            if (isSupported(Parameters.ANTIBANDING_AUTO, mParameters.getSupportedAntibanding())) {
-                mParameters.setAntibanding(Parameters.ANTIBANDING_AUTO);
-            }
-        } else {
-            if (isSupported(mDefaultAntibanding, mParameters.getSupportedAntibanding())) {
-                mParameters.setAntibanding(mDefaultAntibanding);
-            }
-        }
-        Log.d(TAG, "antiBanding value = " + mParameters.getAntibanding());
-
         mUnsupportedHFRVideoSize = false;
         mUnsupportedHFRVideoCodec = false;
         mUnsupportedHSRVideoSize = false;
-- 
2.17.1

